presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -454,8 +455,14 @@ else if (partitioningHandle.equals(SCALED_WRITER_DISTRIBUTION)) {
                     // verify execution is consistent with planner's decision on dynamic lifespan schedule
                     verify(bucketNodeMap.isDynamic() == dynamicLifespanSchedule);
 
-                    stageNodeList = new ArrayList<>(nodeScheduler.createNodeSelector(connectorId).allNodes());
-                    Collections.shuffle(stageNodeList);
+                    if (!bucketNodeMap.isDynamic()) {
```

Comment:
Why not just

```
NodeSelector nodeSelector = nodeScheduler.createNodeSelector(connectorId);
                    if (!bucketNodeMap.isDynamic()) {
                        stageNodeList = nodeSelector.allNodes();
                    }
                    else {
                        stageNodeList = nodeSelector.selectRandomNodes(getMaxTasksPerStage(session));
                    }
```

Reply:
@arhimondr : Then you are creating tasks on all the nodes when `bucketNodeMap` is static right? (And some tasks will not have any splits...)

Reply:
Yeah, nevermind =)